src/include/misc_inline.h

```diff
@@ -315,6 +315,24 @@ __wt_read_shared_double(double *to_read)
     return (*to_read);
 }
 
+/*
+ * __wt_set_shared_maximum_u64 --
+ *     Given an unsigned integer in memory overwrite it with a new value if it is larger.
+ */
+static WT_INLINE void
+__wt_set_shared_maximum_u64(uint64_t *shared_value, uint64_t new_val)
+{
+    uint64_t current;
+
+    current = *shared_value;
```

Comment:
Do either of loads create an additional data race or does the CAS prevent that? Applying the C standard definition of a "data-race"
> The execution of a program contains a data race if it contains two potentially concurrent conflicting actions, at least one of which is not atomic, and neither happens before the other, except for the special case for signal handlers described below [omitted]. Any such data race results in undefined behavior.

Given the CAS orders the first load before and subsequent loads after I think we are safe.